projects/ngx-darkbox-gallery/src/lib/ngx-darkbox-gallery.component.ts

```diff
@@ -153,37 +179,36 @@ export class NgxDarkboxGalleryComponent implements OnInit, OnChanges {
    * @return the new index
    */
   private calculateValidImageIndex(increase: boolean): number {
-    const loopDirection = this.effectiveConfiguration.darkboxConfiguration.loopDirection;
-    const addend = increase ? 1 : -1;
-    const targetIndex = this.currentImageIndex + addend;
-    const maxImageIndex = this.images.length - 1;
-
-    if (targetIndex >= 0 && targetIndex <= maxImageIndex) {
+    const newImageIndex = this.imageIndexService.calculateValidImageIndex(this.effectiveConfiguration, this.currentImageIndex, this.images.length, increase);
+    const rollOverRequired = newImageIndex >= this.images.length - 1;
```

Comment:
> This condition is never true if the imageService is working correctly. Therefore a rollOver never happens even if it should.
The check works as it is now because of the offset of newImageIndex and this.images.length, but you are right that the `neImageIndex` can never be more than `this.images.length`. I will test the new condition and change it if it works.

Reply:
Sorry, just payed attention to the > not the =.
Condition should work but my suggestion might be a little clearer

Reply:
Since both are working I used the suggested one.